### PR TITLE
Merge RUN_E2E_TESTS and RUN_KITCHEN_TESTS variables

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Second version of this PR: https://github.com/DataDog/datadog-agent/pull/23139

The aim is to merge `RUN_KITCHEN_TESTS` and `RUN_E2E_TESTS` variables

Tested the behavior in the MQ: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/29899192

### Motivation

Make pipeline simpler, and prepare usage of smart unit tests

### CODE REVIEW

**Comment by KevinFairise2** [comment]
/trigger-ci --variable DEPLOY_AGENT=false --variable RUN_KITCHEN_TESTS=false --variable RUN_E2E_TESTS=off

**Comment by KevinFairise2** [comment]
/trigger-ci --variable DEPLOY_AGENT=false --variable RUN_KITCHEN_TESTS=false --variable RUN_E2E_TESTS=of

**Comment by KevinFairise2** [comment]
/trigger-ci --variable DEPLOY_AGENT=false --variable RUN_KITCHEN_TESTS=false --variable RUN_E2E_TESTS=off --variable CI_BRANCH_NAME=main

**Comment by KevinFairise2** [comment]
/trigger-ci --variable DEPLOY_AGENT=false --variable RUN_KITCHEN_TESTS=false --variable RUN_E2E_TESTS=off

**Comment by KevinFairise2** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_E2E_TESTS=on

**Review by spencergilbert — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1068,11 +1059,11 @@ workflow:
       compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
 
 .always_on_container_or_e2e_changes_or_manual:
-  - <<: *if_disable_e2e
+  - <<: *if_disable_e2e_tests
     when: never
   - <<: *if_mergequeue
     when: never
-  - <<: *if_auto_e2e
+  - <<: *if_run_all_e2e_tests
```
Comment (chouetz) [inline .gitlab-ci.yml:1066]: Can you tell me why you don't replace if_auto_e2e by if_auto_e2e_tests here?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -282,31 +282,26 @@ variables:
...
-# Some of the kitchen tests are run on all pipelines by default. They can only be disabled
-# by setting RUN_KITCHEN_TESTS to false.
-.if_default_kitchen: &if_default_kitchen
-  if: $RUN_KITCHEN_TESTS != "false"
+# - on main and deploy pipelines, installer tests are run
+# - on branch pipelines, installer tests are run on a subset of the OSes we test
+# RUN_E2E_TESTS can be set to on to force all the installer tests to be run on a branch pipeline.
+# RUN_E2E_TESTS can be set to false to force installer tests to not run on main/deploy pipelines.
+.if_installer_tests: &if_installer_tests
+  if: ($CI_COMMIT_BRANCH == "main"  || $DEPLOY_AGENT == "true" || $RUN_E2E_TESTS == "on") && $RUN_E2E_TESTS != "off"
```
[the remark this answers is not in the document]
Comment (KevinFairise2) [reply .gitlab-ci.yml:292]: I think it is expected yes, `if_installer_tests` is supposed to replace the `if_kitchen` rule that was true when we should run all the `installer/kitchen` tests. When `if_installer_tests` evaluate to `true` we run all the tests. To run only a subset of `installer` tests on PR we use `.on_default_new-e2e_tests_a7`rule.

**Review by CelianR — COMMENTED** [review]
Reviewed, lgtm 👍 (I don't approve since I don't have the experience yet imo ^^)

**Pull request merged**